ckeditor/widgets.py

```diff
@@ -110,18 +112,16 @@ def render(self, name, value, attrs=None, renderer=None):
             for a, b, c in self.external_plugin_resources
         ]
 
-        return mark_safe(
-            renderer.render(
-                "ckeditor/widget.html",
-                {
-                    "final_attrs": flatatt(final_attrs),
-                    "value": conditional_escape(force_str(value)),
-                    "id": final_attrs["id"],
-                    "config": json_encode(self.config),
-                    "external_plugin_resources": json_encode(external_plugin_resources),
-                },
-            )
-        )
+        context = {
+            "final_attrs": flatatt(final_attrs),
+            "value": conditional_escape(force_str(value)),
+            "id": final_attrs["id"],
+            "config": json_encode(self.config),
+            "external_plugin_resources": json_encode(external_plugin_resources),
+        }
+
+        return self._render(self.template_name, context, renderer)
```

Comment:
https://github.com/django/django/blob/0ab58c120939093fea90822f376e1866fc714d1f/django/forms/widgets.py#L244-L247

You should be able to remove the `def render` method and move the context creation into a `get_context` method.

Reply:
Good call.  Thanks.